### PR TITLE
Lesson 7 Assignment

### PR DESCRIPTION
**Pull request by Swt16**
Lesson 7 Assignment files.

### CODE REVIEW

File: students/ShinTran/lesson07/html_render.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+# This is the framework for the base class
+class Element(object):
+
+    _Indent = '    '    # Default indent of 4 spaces
+    tag = 'html'        # Default tag name set to 'html'
+
+    # Initializes the program, if there's no content, returns an empty list
+    # the style parameter is for formatting a specific tag 
+    def __init__(self, content=None, **style):
```
Comment (charlesrobison) [inline students/ShinTran/lesson07/html_render.py]: General convention is **kwargs for keyword arguments; `def __init__(self, content=None, **kwargs):`

File: students/ShinTran/lesson07/html_render.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+# E.g., <h9> content</h9>
+class H(OneLineElement):
+    tag = 'h'
+    def __init__(self, head_int, content):
+        self.tag = 'h' + str(head_int)
+        super().__init__(content)
+
+# Meta tag that extends SelfClosing
+class Meta(SelfClosing):
+    tag = 'meta'
```
Comment (charlesrobison) [inline students/ShinTran/lesson07/html_render.py]: Produces correct result - nice!

File: students/ShinTran/lesson07/test_html_render.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    body.append(brk1)
+    body.append(brk2)
+    page.append(body)
+    file_contents = render_result(page).strip()
+    assert("<br>") not in file_contents
+    assert("<hr>") not in file_contents
+    lines = file_contents.split('\n')
+    assert lines[3] == '        <meta charset="UTF-8" />'
+    assert lines[11] == "        <hr />"
+    assert lines[15] == "        <br />"
```
Comment (charlesrobison) [inline students/ShinTran/lesson07/test_html_render.py]: Good tests - all pass.

File: students/ShinTran/lesson07/html_render.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+#!/usr/bin/env python3
+
+# A class-based system for rendering html.
+
+
+# This is the framework for the base class
+class Element(object):
+
+    _Indent = '    '    # Default indent of 4 spaces
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson07/html_render.py]: variable convention is snake_case -> `indent`

File: students/ShinTran/lesson07/html_render.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    tag = 'html'        # Default tag name set to 'html'
+
+    # Initializes the program, if there's no content, returns an empty list
+    # the style parameter is for formatting a specific tag 
+    def __init__(self, content=None, **style):
+        self.style = style
+        if content != None:
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson07/html_render.py]: use `is` when comparing objects: `is None`, `is True`

in this case `is not None`

File: students/ShinTran/lesson07/html_render.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+# Prints out a Doctype tag at the beginning of the output file 
+class Html(Element):
+    tag = 'html'
+    def render(self, out_file, indent = ''):
+        out_file.write("<!DOCTYPE html>\n")
+        super().render(out_file, indent)
+
+# Body tag that extends Element
+class Body(Element):
+    tag = 'body'
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson07/html_render.py]: use docstrings for describing functions/classes:
```
class Body(Element):
    """Body tag extends Element"
    pass
```

**Pull request merged**